File: chazm-model/src/main/java/runtimemodels/chazm/model/event/RoleEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.entity.Role;

/**
 * The {@linkplain RoleEventFactory} interface defines the API for constructing {@linkplain RoleEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface RoleEventFactory {

    /**
     * Constructs an {@linkplain RoleEvent}.
     *
     * @param category the {@linkplain EventType}.
     * @param role     the {@linkplain Role}.
     * @return a {@linkplain RoleEvent}.
     */
    RoleEvent build(EventType category, Role role);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/RoleEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.entity.Role

/**
 * The [RoleEventFactory] interface defines the API for constructing [RoleEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface RoleEventFactory {

    /**
     * Constructs an [RoleEvent].
     *
     * @param category the [EventType].
     * @param role     the [Role].
     * @return a [RoleEvent].
     */
    fun build(category: EventType, role: Role): RoleEvent

}
